### PR TITLE
feat: on cluster build doesn't require privileged cluster permissions

### LINKED ISSUE #848: On-cluster build: Implement `build` Task that doesn't require privileged cluster permissions or root

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubalik@gmail.com>

# Changes

- 🎁 on cluster build doesn't require privileged cluster permissions
- 🎁 uses custom version of `buildpacks` Tekton task -> there is added section in `results` step, that stores produced image name and sha to `func.yaml`.

Fixes #848

### CODE REVIEW

File: docs/guides/on-cluster-build.md [hunk 1]
```diff
@@ -57,7 +61,8 @@ git push origin main
 ```bash
 kn func deploy
 ```
-If everything goes fine, you will prompted to provide credentials for the remote container registry that hosts the Function image. You should see output similar to the following:
+Should you are not logged in the container registry referenced in your function configuration,
```
Comment (lance) [inline docs/guides/on-cluster-build.md:64]:
```suggestion
If you are not logged in the container registry referenced in your function configuration,
```

File: pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml [hunk 2]
```diff
@@ -0,0 +1,195 @@
+---
```
Comment (lance) [inline pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml:1]: I'm assuming this is mostly a clone of the tekton buildpack task right?

File: pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        sed -i "s|^image:.*$|image: $(params.APP_IMAGE)|" "$func_file"
+        echo "Function image name: $(params.APP_IMAGE)"
+
+        sed -i "s/^imageDigest:.*$/imageDigest: $digest/" "$func_file"
+        echo "Function image digest: $digest"
+
+        ############################################
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
```
Comment (lance) [inline pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml:189]: Is this the primary difference?

File: pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml (hunk 2 above)
Comment (zroubalik) [reply pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml:1]: Yeah.

File: pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml (hunk 3 above)
Comment (zroubalik) [reply pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml:189]: Right.

File: docs/guides/on-cluster-build.md (hunk 1 above)
Comment (zroubalik) [reply docs/guides/on-cluster-build.md]: Fixed, thanks for the suggestion!

**Pull request merged**